src/searchModal.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+import $ from 'jquery';
+import _ from 'lodash';
+import __ from 'i18n';
+import layoutTpl from 'ui/searchModal/tpl/layout';
+import infoMessageTpl from 'ui/searchModal/tpl/info-message';
+import 'ui/searchModal/css/searchModal.css';
+import component from 'ui/component';
+import 'ui/modal';
+import 'ui/datatable';
+
+/**
+ * Creates a search modal instance
+ * @param {object} config - search modal configuration
+ * @param {string} config.query - search query to be set and triggered on component creation
+ * @param {string} config.url - search endpoint
+ * @param {object} config.events - events hub
+ */
+export default function searchModalFactory(config) {
+    const instance = component().setTemplate(layoutTpl);
+    let searchInput = null;
+    let searchButton = null;
+    let clearButton = null;
+    let running = false;
+    instance.on('render', renderModal);
+    instance.on('destroy', destroyModal);
+
+    /**
+     * Creates search modal, inits template selectors, and triggers initial search
+     */
+    function renderModal() {
+        initModal();
+        initUiSelectors();
+        searchButton.trigger('click');
+    }
+
+    /**
+     * Removes search modal
+     */
+    function destroyModal() {
+        instance.getElement().removeClass('modal').modal('destroy');
+        $('.modal-bg').remove();
```

Comment:
Should we remove `searchButton`, `clearButton` as well?

Reply:
I think it is not necessary. Both buttons, same as other html elements, are inside `.modal-bg`, so should be removed from DOM too, correct?

Reply:
Maybe 🤷‍♂️ that's why I'm asking 🙂. 
Is it possible to call the `renderModal` method twice without page reload?
In case it's possible, could you check that `search` and `clear` executes only one time per click event, otherwise it fine?
```
      searchButton.on('click', search);
      clearButton.on('click', clear);
```

Reply:
Yep, I confirm that is only executed once even if I render the search component multiple times : )